### PR TITLE
feat: improve spyre logits processors for CB

### PR DESCRIPTION
**Pull request by wallashss**
# Description

This PR includes builtin logits processors for spyre. These logits processors are the same from vllm but more optimized to not be wrapped by the `LogitsProcessorWrapper` that will slice logits at each engine step. These logits processors work by calling the `set_prefill_index` and properly handling prefill in our spyre model runner. This PR also extends tests of sampling params to run with continuous batching for the parameters that use logits processors under the hood.

### CODE REVIEW

**Comment by wallashss** [comment]
bot:test

File: vllm_spyre/v1/sample/spyre_logits_processor.py [hunk 1]
```diff
@@ -101,3 +128,127 @@ def apply(self, logits: torch.Tensor) -> torch.Tensor:
...
+        # Adjust min_p
+        adjusted_min_p = max_probabilities.mul_(
+            self.min_p[self._prefill_index].unsqueeze(0))
+        # Identify valid tokens using threshold comparison
+        invalid_token_mask = probability_values < adjusted_min_p
+        # Apply mask using boolean indexing
+        logits[invalid_token_mask] = -float('inf')
+        self._prefill_index = None
+
+        return logits
```
[the remark this answers is not in the document]
Comment (tjohnson31415) [reply vllm_spyre/v1/sample/spyre_logits_processor.py]: > but it only works if the state between prefill and decode are independent

I think a well-behaved LogitsProcessor doesn't need persistent state between prefill and decode, i.e. it can be created from a request after output tokens have been generated. This is needed to support resuming generation after preemption. Could GTI be updated to support this as well?

**Comment by wallashss** [comment]
bot:test

**Comment by wallashss** [comment]
bot:test

**Review by tjohnson31415 — CHANGES_REQUESTED** [review]
@maxdebayser found that this PR has a bug that can be reproduced using the approach in https://github.com/vllm-project/vllm-spyre/issues/508
